modules/ROOT/pages/dialog-components.adoc

```diff
@@ -539,6 +539,101 @@ A *textarea* is a multiline text field.
 }
 ----
 
+[[tree]]
+=== tree
+
+A *tree* is a composite component that presents an expandable category tree expandable down one level. It can also include a search and filter field.
+
+*Events:* Interacting with a *tree* component fires the `+onChange+` function in the dialog’s configuration.
+
+[source,js]
+----
+tinymce.init({
+  selector: "textarea",
+  toolbar: "tree",
+    setup: (ed) => {
+      ed.ui.registry.addButton('tree', {
+      text: 'Tree toolbar button',
+      onAction: () => {
+      ed.windowManager.open({
+        title: 'Tree dialog',
+        buttons: [
+          {
+            type: 'cancel',
+            text: 'Cancel',
+          }
+        ],
+        body : {
+          type: 'panel',
+          items: [{
+            type: 'tree',
+            onLeafAction: (id) => {
+              console.log('clicked on item with id', id);
+            },
+            items: [
+                    {
+      type: 'directory',
+      id: 'dir',
+      title: 'Directory',
+      menu: {
+        icon: 'image-options',
+          type: 'menubutton',
+          fetch: (success) => success([
+            {
+              type: 'menuitem',
+              text: 'Menu item',
+              onAction: () => {
+                store.add('menuitem');
+              }
+            }
+          ])
+      },
+      children: [
+        {
+          type: 'leaf',
+          title: 'Leaf 3',
+          id: '3',
+        },
+        {
+          type: 'leaf',
+          title: 'Leaf 4',
+          id: '4',
+          menu: {
+            icon: 'image-options',
+            type: 'menubutton',
+            fetch: (success) => success([
+              {
+                type: 'menuitem',
+                text: 'Menu item',
+                onAction: () => {
+                  store.add('menuitem');
+                }
+              }
+            ])
+          }
+        },
+      ]
+    },
+    {
+      type: 'leaf',
+      title: 'Leaf 5',
+      id: '5',
+    },
+    {
+      type: 'leaf',
+      title: 'Leaf 6',
+      id: '6',
+    },
+    ]
+    }]
```

Comment:
This needs improved indentation to render correctly

Reply:
Re-formatted this example.